Sources/SwiftParser/StringLiterals.swift

```diff
@@ -44,6 +148,243 @@ extension Parser {
+
+      // The closing delimiter doesn't start on a new line and thus it doesn't
+      // make sense to try and extract indentation from it.
+      return (unexpectedBeforeOpenQuote, openQuote, allSegments, unexpectedBeforeCloseQuote, closeQuote)
+    }
+
+    // -------------------------------------------------------------------------
+    // Parse indentation
+
+    if let lastSegment = lastSegment,
+      let parsedTrivia = parseIndentationTrivia(text: lastSegment.content.tokenText)
+    {
+      indentationTrivia = parsedTrivia
+      indentation = lastSegment.content.tokenText
+      closeQuote = closeQuote.extendingLeadingTrivia(by: parsedTrivia, arena: self.arena)
+    } else {
+      if let lastSegment = lastSegment {
+        indentationTrivia = TriviaParser.parseTrivia(lastSegment.content.tokenText, position: .leading).prefix(while: { $0.isIndentationWhitespace })
+        let indentationByteLength = indentationTrivia.reduce(0, { $0 + $1.byteLength })
+        indentation = SyntaxText(rebasing: lastSegment.content.tokenText[0..<indentationByteLength])
+        middleSegments.append(.stringSegment(lastSegment))
+      } else {
+        indentationTrivia = []
+        indentation = ""
+      }
+
+      unexpectedBeforeCloseQuote = [closeQuote]
+      closeQuote = RawTokenSyntax(missing: closeQuote.tokenKind, leadingTriviaPieces: [.newlines(1)] + indentationTrivia, arena: self.arena)
+    }
+
+    // -------------------------------------------------------------------------
+    // Check open quote followed by newline
+
+    // Condition for the loop below that indicates whether the segment we are
+    // iterating over is on a new line.
+    var isSegmentOnNewLine: Bool
+
+    if let newlineTrivia = firstSegment?.content.tokenText.triviaPieceIfNewline {
+      openQuote = openQuote.extendingTrailingTrivia(by: [newlineTrivia], arena: self.arena)
+      isSegmentOnNewLine = true
+    } else {
+      if let firstSegment = firstSegment {
+        middleSegments.insert(.stringSegment(firstSegment), at: 0)
+      }
+      isSegmentOnNewLine = false
+      unexpectedBeforeOpenQuote = [openQuote]
+      openQuote = RawTokenSyntax(missing: openQuote.tokenKind, trailingTriviaPieces: [.newlines(1)] + indentationTrivia, arena: self.arena)
+    }
+
+    // -------------------------------------------------------------------------
+    // Check indentation of segments and escaped newlines at end of segment
+
+    let expressionIndentationChecker = StringLiteralExpressionIndentationChecker(expectedIndentation: indentation, arena: self.arena)
+
+    for (index, segment) in middleSegments.enumerated() {
+      switch segment {
+      case .stringSegment(var segment):
+        // We are not considering leading trivia for indentation computation.
+        // If these assertions are violated, we can probably lift them but we
+        // would need to check the produce the expected results.
```

Comment:
```
        // would need to check the produce the expected results.
```

`to check the produce` 😅. `we` maybe? IMO I'd probably just skip the comment and the assertion since we already have this assertion at the start of the function (for all segments).

Reply:
This one is particularly checking that we haven’t introduced leading trivia in one of the post-processing steps before.

I originally also had an assertion that we don’t have any unexpected text here and needed to lift that, which was OK, it just reminded me that something changed.